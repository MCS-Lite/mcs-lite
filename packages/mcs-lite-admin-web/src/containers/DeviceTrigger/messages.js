import { defineMessages } from 'react-intl';

const messages = defineMessages({
  'DeviceTrigger.triggerAndAction': {
    id: 'DeviceTrigger.triggerAndAction',
    description: 'Title',
    defaultMessage: '觸發條件與動作',
  },
  'DeviceTrigger.on': {
    id: 'DeviceTrigger.on',
    defaultMessage: 'ON',
  },
  'DeviceTrigger.off': {
    id: 'DeviceTrigger.off',
    defaultMessage: 'OFF',
  },
});

export default messages;
