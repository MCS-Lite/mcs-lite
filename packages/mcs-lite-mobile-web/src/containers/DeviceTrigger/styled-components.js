import styled from 'styled-components';
import Small from 'mcs-lite-ui/lib/Small';
import MobileContentWrapper from 'mcs-lite-ui/lib/MobileContentWrapper';

export const Item = styled(MobileContentWrapper)`
  padding: 8px 16px;
  border-bottom: 1px solid ${props => props.theme.color.grayBase};
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const StyledSamll = styled(Small)`
  display: block;
  margin-top: 4px;
  color: ${props => props.theme.color.grayDark};
`;

export const IconWrapper = styled.div`
  font-size: 24px;
  color: ${props => props.theme.color.primary};
`;
