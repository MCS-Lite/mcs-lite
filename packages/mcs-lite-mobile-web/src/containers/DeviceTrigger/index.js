import { connect } from 'react-redux';
import compose from 'recompose/compose';
import { withGetMessages } from 'react-intl-inject-hoc';
import messages from './messages';
import { actions } from '../../modules/devices';
import DeviceTrigger from './DeviceTrigger';

export const mapStateToProps = ({ devices, ui }, { params: { deviceId } }) => ({
  deviceId,
  device: devices[deviceId],
  isLoading: ui.isLoading,
});
export const mapDispatchToProps = {
  fetchDeviceDetail: actions.fetchDeviceDetail,
};

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withGetMessages(messages, 'DeviceTrigger')
)(DeviceTrigger);
