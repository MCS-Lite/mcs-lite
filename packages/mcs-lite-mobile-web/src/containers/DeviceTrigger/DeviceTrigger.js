import React from 'react';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import IconChevronRight from 'mcs-lite-icon/lib/IconChevronRight';
import { PreventDrag, PullToRefresh, P } from 'mcs-lite-ui';
import compose from 'recompose/compose';
import withGetMessages from '../../utils/withGetMessages';
import messages from './messages';
import { actions } from '../../modules/devices';
import Header from '../../components/Header';
import StyledLink from '../../components/StyledLink';
import { Item, StyledSamll, IconWrapper } from './styled-components';
import updatePathname from '../../utils/updatePathname';

class DeviceTrigger extends React.Component {
  state = { isMenuShow: false, target: undefined };
  componentDidMount = () => this.fetch();
  fetch = () => this.props.fetchDeviceDetail(this.props.deviceId);
  render() {
    const { device, isLoading, getMessages: t } = this.props;
    const { fetch } = this;

    return (
      <div>
        <Helmet title={t('triggerAndAction')} />
        <Header title={t('triggerAndAction')} backTo={updatePathname(`/devices/${device && device.deviceId}`)} />
        <main>
          <PullToRefresh isLoading={isLoading} onPull={fetch}>
            <PreventDrag>
              <StyledLink to={updatePathname(`/devices/${device && device.deviceId}/trigger/edit`)}>
                <Item>
                  <div>
                    <P>觸發條件名稱 A</P>
                    <StyledSamll>{t('on')}</StyledSamll>
                  </div>
                  <IconWrapper><IconChevronRight /></IconWrapper>
                </Item>
              </StyledLink>

              <StyledLink to={updatePathname(`/devices/${device && device.deviceId}/trigger/edit`)}>
                <Item>
                  <div>
                    <P>觸發條件名稱 A</P>
                    <StyledSamll>{t('off')}</StyledSamll>
                  </div>
                  <IconWrapper><IconChevronRight /></IconWrapper>
                </Item>
              </StyledLink>
            </PreventDrag>
          </PullToRefresh>
        </main>
      </div>
    );
  }
}

export default compose(
  connect(
    ({ devices, ui }, { params: { deviceId }}) => ({
      deviceId,
      device: devices[deviceId],
      isLoading: ui.isLoading,
    }),
    { fetchDeviceDetail: actions.fetchDeviceDetail },
  ),
  withGetMessages(messages, 'DeviceTrigger'),
)(DeviceTrigger);
