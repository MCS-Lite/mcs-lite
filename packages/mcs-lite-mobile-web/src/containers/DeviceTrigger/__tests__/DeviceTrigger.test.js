import React from 'react';
import * as R from 'ramda';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DeviceTrigger from '../DeviceTrigger';

it('should renders <DeviceTrigger> correctly', () => {
  const fetchMock = jest.fn();
  const wrapper = shallow(
    <DeviceTrigger
      getMessages={R.identity}
      deviceId="deviceId"
      device={{
        deviceId: 'deviceId',
        deviceName: 'deviceName',
        createUserId: 'createUserId',
        deviceDescription: 'deviceDescription',
        deviceKey: 'deviceKey',
      }}
      isLoading={false}
      fetchDeviceDetail={fetchMock}
    />,
  );

  expect(toJson(wrapper)).toMatchSnapshot();
  expect(fetchMock).toHaveBeenCalledWith('deviceId');
});
