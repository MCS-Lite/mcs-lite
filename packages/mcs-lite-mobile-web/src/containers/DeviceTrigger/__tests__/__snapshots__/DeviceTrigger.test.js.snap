// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders <DeviceTrigger> correctly 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      triggerAndAction
    </title>
  </HelmetWrapper>
  <MobileHeader
    leftChildren={
      <MobileHeaderIcon
        component={[Function]}
        to={[Function]}
      >
        <IconArrowLeft
          size="1em"
        />
      </MobileHeaderIcon>
    }
    title="triggerAndAction"
  />
  <main>
    <PullToRefresh
      IconArrow={
        <Heading
          color="grayDark"
          level={2}
        >
          <IconGoDown
            size="1em"
          />
        </Heading>
      }
      IconLoading={
        <Heading
          color="grayDark"
          level={2}
        >
          <Spin>
            <IconLoading
              size="1em"
            />
          </Spin>
        </Heading>
      }
      isLoading={false}
      onPull={[Function]}
    >
      <PreventDrag>
        <StyledLink
          to={[Function]}
        >
          <Styled(MobileContentWrapper)>
            <div>
              <P>
                觸發條件名稱 A
              </P>
              <Styled(Small)>
                on
              </Styled(Small)>
            </div>
            <styled.div>
              <IconChevronRight
                size="1em"
              />
            </styled.div>
          </Styled(MobileContentWrapper)>
        </StyledLink>
        <StyledLink
          to={[Function]}
        >
          <Styled(MobileContentWrapper)>
            <div>
              <P>
                觸發條件名稱 A
              </P>
              <Styled(Small)>
                off
              </Styled(Small)>
            </div>
            <styled.div>
              <IconChevronRight
                size="1em"
              />
            </styled.div>
          </Styled(MobileContentWrapper)>
        </StyledLink>
      </PreventDrag>
    </PullToRefresh>
  </main>
</div>
`;
